src/parse/createFieldParser.ts

```diff
@@ -41,32 +76,43 @@ module.exports = function createFieldParser(fieldOptions) {
         end: range.end,
         raw: getText(lines, range),
       })),
+      line: 0,
+      start: 0,
+      end: 0,
     };
     const range = result.ranges[0];
     result.line = range.line;
     result.start = range.start;
     result.end = range.end;
     try {
-      let parsed = fieldOptions.parser(source, ...related);
+      const parsed = fieldOptions.parser(source, ...textRelated);
       result.value = typeof parsed === 'object' ? parsed.value : parsed;
       result.valid = true;
       if (typeof parsed === 'object') {
         result.start = range.start + parsed.start;
         result.end = range.start + parsed.end;
       }
     } catch (e) {
-      result.error = e.message;
+      if (typeof e === 'string') {
+        result.error = e;
```

Comment:
this is weird. we should never throw strings

Reply:
in our case yes.
i made this to make the general case that's all

Reply:
please keep it simple. we have rules that prevent throwing strings.